.github/workflows/CI.yml

```diff
@@ -350,22 +350,67 @@ jobs:
           key: "cached_image_${{ github.run_id }}"
           path: "${{ env.PACKAGING_ROOT }}"
 
-      - name: restore cached binaries
+      - name: Restore cached binaries
         if: steps.cached_image.outputs.cache-hit != 'true'
         uses: actions/cache@v2
         with:
           key: "cached_binaries_${{ github.run_id }}"
           path: build
 
-      - name: Build Image
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Make package root directory
+        if: steps.cached_image.outputs.cache-hit != 'true'
+        run: mkdir -p $PACKAGING_ROOT
+
+      #Build the adot collector image for two primary reasons:
+      #-Using the adot collector image to do the integration test
+      #-Export it for delivery version image in CD
+      #Documentation: https://github.com/docker/build-push-action
+      - name: Build adot collector image
+        id: docker_build
+        uses: docker/build-push-action@v2
         if: steps.cached_image.outputs.cache-hit != 'true'
-        run: docker build -t $IMAGE_NAME -f cmd/awscollector/Dockerfile .
+        with:
+          dockerfile: cmd/awscollector/Dockerfile
+          context: .
+          load: true #Export as a local image to push it to ECR since we need to export as tar and cache it.
+          tags: $IMAGE_NAME:latest
+          outputs: type=docker,dest=$PACKAGING_ROOT/$IMAGE_NAME.tar
 
-      - name: Extract the Image file
+      - name: Login to Public ECR
+        if: steps.cached_image.outputs.cache-hit != 'true'
+        id: login-ecr
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
+          password: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
+        env:
+          AWS_REGION: us-east-1
+
+      - name: upload to ECR
         if: steps.cached_image.outputs.cache-hit != 'true'
         run: |
-          mkdir -p $PACKAGING_ROOT
-          docker save --output $PACKAGING_ROOT/$IMAGE_NAME.tar $IMAGE_NAME
+          docker tag $IMAGE_NAME:latest public.ecr.aws/$ECR_REPO:${{ needs.e2etest-preparation.outputs.version }}
```

Comment:
Can just use the action to do the tagging/pushing.

Reply:
True. I was testing different but was not able to do it with the sub_branch. So I changed my latest commit in another branch and test it there.